pkg/power/accelerator/qat/source/qat_utils.go

```diff
@@ -0,0 +1,201 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Copyright 2023.
+*/
+
+package source
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	//telemetry base path
+	teleBasePath = "/sys/devices/pci0000:%s/0000:%s:00.0/telemetry/"
+
+	//control telemetry swtich path
+	controlPath = filepath.Join(teleBasePath, "control")
+
+	//obtain device utilization data path
+	deviceDataPath = filepath.Join(teleBasePath, "device_data")
+)
+
+// getDevices obtain available qat devices and search for ID
+func getDevices() (map[string]interface{}, error) {
+	//Available devices
+	availableDev := make(map[string]interface{})
+
+	//use adf_ctl get qat devices status
+	commandText := "adf_ctl status"
+	cmd := exec.Command("bash", "-c", commandText)
+	out, err := cmd.CombinedOutput()
```

Comment:
can you add unit test here to validate the parsing? You can do it in a follow up PR.